### PR TITLE
Upgraded personalization api to v2

### PR DESCRIPTION
**Pull request by dzannotti**
as per @frsyuki request

### CODE REVIEW

File: test/fixtures/usersegment/index.html [hunk 1]
```diff
@@ -65,11 +65,26 @@
...
 
 
+    var td = new Treasure({
+      database: 'test_db_request',
+      writeKey: '91/96da3cfb876cc50724d0dddef670d95eea2a0018',
+      host: 'in-staging.treasuredata.com'
+    })
+    td.fetchUserSegments('55198252-3752-4a1f-b011-62a821cae61d', function (key, values, attributes) {
+      if (key !== undefined && values !== undefined && attributes !== undefined) {
+        setStatus(1, 'success')
```
Comment (danreeves) [inline test/fixtures/usersegment/index.html]: should be `setStatus(2`

File: README.md [hunk 2]
```diff
@@ -300,15 +300,25 @@ td.fetchGlobalID(successCallback, errorCallback)
 **Parameters:**
 
 * **token** : String (required) - Audience Token for the userId
-* **success** : Function (optional) - Callback for receiving the user key and segments
+* **success** : Function (optional) - Callback for receiving the user key, attributes an segments
```
Comment (danreeves) [inline README.md]: typo: and

Comment (danreeves) [reply README.md]: This typo is still here

File: test/fixtures/usersegment/index.html [hunk 3]
```diff
@@ -39,8 +39,12 @@
       keys: {
         email: 'someone@somewhere.com'
       }
-    }, function (keys) {
-      if (keys !== undefined) {
+    }, function (response) {
+      if (response &&
+          response.length &&
+          response[0].key !== undefined &&
+          Object.prototype.toString.call(response[0].values) === '[object Array]'
```
Comment (danreeves) [inline test/fixtures/usersegment/index.html]: We're not confirming attributes object is returned

Comment (dzannotti) [reply test/fixtures/usersegment/index.html]: the attributes object is not always returned (depending on input values), and due to the nature of the e2e i didn't feel comfortable checking for it's presence

**Comment by danreeves** [comment]
**NOTE* This is a breaking change and the release version should correspond

**Comment by frsyuki** [comment]
@dzannotti Great. Thank you for your quick work!
This is just an idea but is it possible to keep backward compatibility by checking `successCallback.length == 2` so that we can encourage customers to upgrade quickly? Like this way:

```diff
diff --git a/README.md b/README.md
index ce7dd49..c2e98d2 100644
--- a/README.md
+++ b/README.md
@@ -300,7 +300,7 @@ td.fetchGlobalID(successCallback, errorCallback)
 **Parameters:**
 
 * **token** : String (required) - Audience Token for the userId
-* **success** : Function (optional) - Callback for receiving the user key and segments
+* **success** : Function (optional) - Callback for receiving the user key, attributes and segments
 * **error** : Function (optional) - Callback for when sending the event is unsuccessful
 
 **Example:**
@@ -308,7 +308,26 @@ td.fetchGlobalID(successCallback, errorCallback)
 ```javascript
 var td = new Treasure({...})
 
-var successCallback = function (key, segments) {
+var successCallback = function (result) {
+  /* personalization result api
+     result.keys: {key1: value, key2: value}
+     result.attributes: {age: 30, area: "US"}
+     result.values: ["1234", "5678"]
+     result.objects: [
+       {
+         key: {key1: value},
+         values: ["1234"]
+         attributes: {age: 30}
+         audienceId: "391",
+       },
+       {
+         key: {key2: value},
+         values: ["5678"]
+         attributes: {area: "US"}
+         audienceId: "679",
+       }
+     ]
+  */
   // celebrate();
 };
 
diff --git a/lib/plugins/personalization.js b/lib/plugins/personalization.js
index ba61912..4d56d3f 100644
--- a/lib/plugins/personalization.js
+++ b/lib/plugins/personalization.js
@@ -50,13 +50,46 @@ function fetchUserSegments (tokenOrConfig, successCallback, errorCallback) {
 
   var keyString = keysArray.join('&')
 
-  var url = 'https://' + this.client.cdpHost + '/cdp/lookup/collect/segments?version=1&token=' + token + (keyString && ('&' + keyString))
+  var url = 'https://' + this.client.cdpHost + '/cdp/lookup/collect/segments?version=2&token=' + token + (keyString && ('&' + keyString))
 
   jsonp(url, {
     prefix: 'TreasureJSONPCallback',
     timeout: 10000
   }, function (err, res) {
-    return err ? errorCallback(err) : successCallback(res && (res.key || res.keys), res && res.values)
+    if (err) {
+      return errorCallback(err)
+    } else {
+      var result = {
+        objects: res,
+        get keys() {
+          var merged = {}
+          this.objects.forEach(function (e) {
+            Object.assign(merged, e.key)
+          })
+          return merged
+        },
+        get attributes() {
+          var merged = {}
+          this.objects.forEach(function (e) {
+            Object.assign(merged, e.attributes)
+          })
+          return merged
+        },
+        get values() {
+          var merged = []
+          this.objects.forEach(function (e) {
+            Array.prototype.push.apply(merged, e.values)
+          })
+          return merged
+        }
+      }
+      if (successCallback.length == 2) {
+        // for backward compatibility
+        successCallback(result.keys, result.values)
+      } else {
+        successCallback(result)
+      }
+    }
   })
 }
 
diff --git a/test/fixtures/usersegment/index.html b/test/fixtures/usersegment/index.html
index f61387c..eb382b8 100644
--- a/test/fixtures/usersegment/index.html
+++ b/test/fixtures/usersegment/index.html
@@ -39,8 +39,14 @@
       keys: {
         email: 'someone@somewhere.com'
       }
-    }, function (keys) {
-      if (keys !== undefined) {
+    }, function (response) {
+      if (response &&
+          response.length &&
+          response[0].key !== undefined &&
+          response[0].attributes !== undefined &&
+          response[0].audienceId !== undefined &&
+          Object.prototype.toString.call(response[0].values) === '[object Array]'
+      ) {
         setStatus(0, 'success')
       } else {
         setStatus(0, 'failure')
@@ -54,8 +60,14 @@
       writeKey: '91/96da3cfb876cc50724d0dddef670d95eea2a0018',
       host: 'in-staging.treasuredata.com'
     })
-    td.fetchUserSegments('55198252-3752-4a1f-b011-62a821cae61d', function (key) {
-      if (key !== undefined) {
+    td.fetchUserSegments('55198252-3752-4a1f-b011-62a821cae61d', function (response) {
+      if (response &&
+          response.length &&
+          response[0].key !== undefined &&
+          response[0].attributes !== undefined &&
+          response[0].audienceId !== undefined &&
+          Object.prototype.toString.call(response[0].values) === '[object Array]'
+      ) {
         setStatus(1, 'success')
       } else {
         setStatus(1, 'failure')
@@ -63,8 +75,6 @@
     }, function () {
       setStatus(1, 'failure')
     })
-
-
   })()
 </script>
 
```

File: test/fixtures/usersegment/index.html [hunk 4]
```diff
@@ -39,8 +39,14 @@
...
         email: 'someone@somewhere.com'
       }
-    }, function (keys) {
-      if (keys !== undefined) {
+    }, function (response) {
+      if (response &&
+          response.length &&
+          response[0].key !== undefined &&
+          response[0].attributes !== undefined &&
+          response[0].extraFields !== undefined &&
```
Comment (frsyuki) [inline test/fixtures/usersegment/index.html]: I found that `extraFields` is a bug...Personalization API now returns `extraFields: {audienceId: "999"}` but it should be `audienceId: "999"` in fact. I'll deploy a fix.

Comment (frsyuki) [reply test/fixtures/usersegment/index.html]: https://github.com/treasure-data/td-cdp-kvs-server/pull/108

**Comment by dzannotti** [comment]
@frsyuki thanks! Backwards compatibility isn't necessary as this will be released alongside semver 2.0 to support GDPR

**Pull request merged**